mono/metadata/custom-attrs.c

```diff
@@ -1644,30 +1645,30 @@ mono_custom_attrs_from_index_checked (MonoImage *image, guint32 idx, gboolean ig
 			g_error ("Unknown table for custom attr type %08x", cols [MONO_CUSTOM_ATTR_TYPE]);
 			break;
 		}
-		attr = &ainfo->attrs [i - 1];
+		attr = &ainfo->attrs [i];
 		attr->ctor = mono_get_method_checked (image, mtoken, NULL, NULL, error);
 		if (!attr->ctor) {
 			g_warning ("Can't find custom attr constructor image: %s mtoken: 0x%08x due to: %s", image->name, mtoken, mono_error_get_message (error));
 			if (ignore_missing) {
 				mono_error_cleanup (error);
 				error_init (error);
 			} else {
-				g_list_free (list);
+				g_array_free (attr_array, TRUE);
```

Comment:
Attempt single point of exit?
Or C++ destructors would work well here.

Reply:
Not really worth the trouble imo, given I'm just swapping out the free functions in a couple places.

Reply:
Single point of exit is always a good idea in C imho.
It makes for smaller code, i.e. faster code, and easier to maintain.